algorithms/HPdclassifier/R/hpdegbm.R

```diff
@@ -0,0 +1,288 @@
+# This program is free software; you can redistribute it and/or                     
+# modify it under the terms of the GNU General Public License                       
+# as published by the Free Software Foundation; either version 2                    
+# of the License, or (at your option) any later version.                            
+#                                                                                   
+# This program is distributed in the hope that it will be useful,                   
+# but WITHOUT ANY WARRANTY; without even the implied warranty of                    
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                      
+# GNU General Public License for more details.                                      
+#                                                                                   
+# You should have received a copy of the GNU General Public License                 
+# along with this program; if not, write to the Free Software                       
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.    
+####################################################################################
+
+####################################################################################
+#  File hpdegbm.R
+#  Distributed version of gbm 
+
+#  This code is a distributed version of gbm function availabel in R gbm package.
+
+#  myModel <- hpdegbm (.....)
+#  GBM_model <- myModel[[1]] # trained GBM sub-models
+#  best.iter <- myModel[[2]] # best iteration of each sub-models
+
+####################################################################################
+## A copy of gbm R package license:
+#
+#  Copyright (C) 1995-2012 The R Core Team
+#
+#  This program is free software; you can redistribute it and/or modify
+#  it under the terms of the GNU General Public License as published by
+#  the Free Software Foundation; either version 2 of the License, or
+#  (at your option) any later version.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  A copy of the GNU General Public License is available at
+#  http://www.r-project.org/Licenses/
+
+####################################################################################
+
+
+
+
+####################################################################################
+###                Model training function of distributed GBM
+####################################################################################
+hpdegbm <- function(
+       X_train,Y_train, 
+       dl_GBM_model,
+       dbest.iter, 
+       nExecutor,                                         
+       #distribution = "adaboost",
+       distribution = "bernoulli",
+       n.trees = 1000, 
```

Comment:
Why the default value of "n.trees" is not the same as vanilla gbm? I don't see why we should change it.